src/XCCDF/result.c

```diff
@@ -313,23 +376,28 @@ void xccdf_result_fill_sysinfo(struct xccdf_result *result)
 			if (!ifa->ifa_addr)
 				continue;
 			family = ifa->ifa_addr->sa_family;
-			if (family != AF_INET && family != AF_INET6)
-				continue;
-
-			if (family == AF_INET) {
-				if (getnameinfo(ifa->ifa_addr, sizeof(struct sockaddr_in),
-						hostip, sizeof(hostip), NULL, 0, NI_NUMERICHOST))
-					goto out2;
-			} else {
-				struct sockaddr_in6 *sin6;
 
-				sin6 = (struct sockaddr_in6 *) ifa->ifa_addr;
-				if (!inet_ntop(family, (const void *) &sin6->sin6_addr,
-						   hostip, sizeof(hostip)))
-					goto out2;
+			if (family == AF_INET || family == AF_INET6) {
+				if (family == AF_INET) {
+					if (getnameinfo(ifa->ifa_addr, sizeof(struct sockaddr_in),
+							hostip, sizeof(hostip), NULL, 0, NI_NUMERICHOST))
+						goto out2;
+				} else if (family == AF_INET6) {
+					struct sockaddr_in6 *sin6;
+
+					sin6 = (struct sockaddr_in6 *) ifa->ifa_addr;
+					if (!inet_ntop(family, (const void *) &sin6->sin6_addr,
+							hostip, sizeof(hostip)))
+						goto out2;
```

Comment:
This kind of reminds me the old rule  "if you need more than 3 levels of indentation, you are screwed, and you should fix your program" :smile: In other words, do you think this can be extracted to a separate function?

Reply:
This whole function cries for refactoring. But we have bigger fish to fry at this moment, I was trying to make the least invasive change.